File: js/warriorNicknames.js
var nm1 = ["The Animal","The Anomaly","The Anvil","The Army","The Axe","The Barbarian","The Bear","The Bearclaw","The Beast","The Behemoth","The Blade","The Bloodlust","The Bloody","The Boar","The Boulder","The Boulderfist","The Brute","The Bull","The Butcher","The Challenger","The Champion","The Cobra","The Colossal","The Colossus","The Conqueror","The Corrupter","The Coyote","The Crooked","The Crow","The Cruel","The Cunning","The Cursed","The Dagger","The Danger","The Deceiver","The Defiant","The Delirious","The Demigod","The Demon","The Deserter","The Destroyer","The Devil","The Dire","The Dire Bear","The Dire One","The Dire Wolf","The Dog","The Dragon","The Dragonheart","The Drake","The Dread","The Edge","The Enigma","The Eternal Hunger","The Exalted","The Executioner","The Exile","The Fang","The Fearless","The Feisty","The Feral","The Fiend","The Fierce","The Fire","The Firestarter","The Flame","The Flurry","The Forsaken","The Freak","The Fury","The Ghost","The Giant","The Giantslayer","The Gold One","The Grave","The Gravedigger","The Gravekeeper","The Grim","The Grotesque","The Guardian","The Hallowed","The Hammer","The Hawk","The Heartless","The Hog","The Hollow","The Honorbound","The Honorless","The Horror","The Hound","The Hunger","The Hungry","The Hunter","The Hurricane","The Impure","The Insane","The Invincible","The Ironclad","The Ironfist","The Jackal","The Jester","The Legionnaire","The Limp","The Lion","The Lionheart","The Lone Wolf","The Magnificent","The Mammoth","The Maneater","The Maniac","The Marked","The Marked One","The Martyr","The Menace","The Merciful","The Miracle","The Monster","The Mountain","The Mutant","The Mute","The Nightmare","The Nightowl","The Noble","The Nobody","The Observer","The Owl","The Ox","The Paragon","The Patient","The Phantom","The Phoenix","The Prince","The Prodigy","The Pyro","The Rabid","The Raven","The Rebel","The Reckless","The Rider","The Rogue","The Sadist","The Sadistic","The Savage","The Scar","The Scarred One","The Sentinel","The Serpent","The Shadow","The Shady","The Shepherd","The Silencer","The Silent","The Silver One","The Skeleton","The Slayer","The Sleeper","The Spider","The Spook","The Storm","The Surgeon","The Swine","The Temper","The Tempest","The Terror","The Thief","The Thirst","The Thug","The Thunder","The Titan","The Tower","The Traitor","The Tyrant","The Undying","The Valiant","The Vengeful","The Venom Tongue","The Vermin","The Vicious","The Viper","The Vulture","The Whisper","The Widow Maker","The Wild","The Wildling","The Wolf","The Wonder","The Wyrm"];
var nm2 = ["Battle","Bear","Blood","Bone","Boulder","Bright","Dark","Dead","Death","Demon","Doom","Dragon","Ember","Fire","Fist","Frost","Fuse","Giant","Gold","Gore","Grand","Great","Hell","Iron","Light","Mammoth","Molten","Night","Phoenix","Proud","Rage","Raven","Red","Rock","Rumble","Shadow","Sharp","Shield","Silent","Silver","Single","Skull","Spirit","Steel","Stone","Storm","Stout","Strong","Swift","Thunder","True","Void","War","Wild","Wolf"];
var nm3 = ["bane","blade","blood","blow","bolt","bow","breaker","brow","chaser","claw","cleaver","crest","cut","eye","fang","fist","flayer","fury","gaze","grim","grimace","grip","hair","hallow","hammer","hand","head","heart","helm","hide","mane","mantle","might","pelt","rage","roar","scar","scream","shade","shadow","shield","shout","snarl","song","sorrow","stare","stride","strike","sword","sworn","talon","thorn","tongue","visage"];
var nm4 = ["",""," "];
var br = "";

function nameGen(type){
	$('#placeholder').css('textTransform', 'capitalize');
	var tp = type;
	var element = document.createElement("div");
	element.setAttribute("id", "result");
	
	for(i = 0; i < 10; i++){
		if(i < 5){
			rnd = Math.floor(Math.random() * nm1.length);
			names = nm1[rnd];
		}else{
			rnd = Math.floor(Math.random() * nm2.length);
			rnd2 = Math.floor(Math.random() * nm3.length);
			rnd3 = Math.floor(Math.random() * nm4.length);
			names = nm2[rnd] + nm4[rnd3] + nm3[rnd2];
		}
		br = document.createElement('br');	
		element.appendChild(document.createTextNode(names));
		element.appendChild(br);
	}
	if(document.getElementById("result")){
		document.getElementById("placeholder").removeChild(document.getElementById("result"));
	}		
	document.getElementById("placeholder").appendChild(element);
}

File: generators/warriorNicknames.py
__all__ = ['warriorNicknames']

# Don't look below, you will not understand this Python code :) I don't.

from js2py.pyjs import *
# setting scope
var = Scope( JS_BUILTINS )
set_global_object(var)

# Code follows:
var.registers(['nm1', 'nm2', 'nm3', 'nm4', 'nameGen'])
@Js
def PyJsHoisted_nameGen_(type, this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this, 'type':type}, var)
    var.registers(['tp', 'result', 'type'])
    var.put('tp', var.get('type'))
    var.put('result', Js([]))
    #for JS loop
    var.put('i', Js(0.0))
    while (var.get('i')<Js(10.0)):
        try:
            if (var.get('i')<Js(5.0)):
                var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm1').get('length'))))
                var.put('names', var.get('nm1').get(var.get('rnd')))
            else:
                var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
                var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm3').get('length'))))
                var.put('rnd3', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm4').get('length'))))
                var.put('names', ((var.get('nm2').get(var.get('rnd'))+var.get('nm4').get(var.get('rnd3')))+var.get('nm3').get(var.get('rnd2'))))
            var.get('result').callprop('push', var.get('names'))
        finally:
                (var.put('i',Js(var.get('i').to_number())+Js(1))-Js(1))
    pass
    return var.get('result')
PyJsHoisted_nameGen_.func_name = 'nameGen'
var.put('nameGen', PyJsHoisted_nameGen_)
var.put('nm1', Js([Js('The Animal'), Js('The Anomaly'), Js('The Anvil'), Js('The Army'), Js('The Axe'), Js('The Barbarian'), Js('The Bear'), Js('The Bearclaw'), Js('The Beast'), Js('The Behemoth'), Js('The Blade'), Js('The Bloodlust'), Js('The Bloody'), Js('The Boar'), Js('The Boulder'), Js('The Boulderfist'), Js('The Brute'), Js('The Bull'), Js('The Butcher'), Js('The Challenger'), Js('The Champion'), Js('The Cobra'), Js('The Colossal'), Js('The Colossus'), Js('The Conqueror'), Js('The Corrupter'), Js('The Coyote'), Js('The Crooked'), Js('The Crow'), Js('The Cruel'), Js('The Cunning'), Js('The Cursed'), Js('The Dagger'), Js('The Danger'), Js('The Deceiver'), Js('The Defiant'), Js('The Delirious'), Js('The Demigod'), Js('The Demon'), Js('The Deserter'), Js('The Destroyer'), Js('The Devil'), Js('The Dire'), Js('The Dire Bear'), Js('The Dire One'), Js('The Dire Wolf'), Js('The Dog'), Js('The Dragon'), Js('The Dragonheart'), Js('The Drake'), Js('The Dread'), Js('The Edge'), Js('The Enigma'), Js('The Eternal Hunger'), Js('The Exalted'), Js('The Executioner'), Js('The Exile'), Js('The Fang'), Js('The Fearless'), Js('The Feisty'), Js('The Feral'), Js('The Fiend'), Js('The Fierce'), Js('The Fire'), Js('The Firestarter'), Js('The Flame'), Js('The Flurry'), Js('The Forsaken'), Js('The Freak'), Js('The Fury'), Js('The Ghost'), Js('The Giant'), Js('The Giantslayer'), Js('The Gold One'), Js('The Grave'), Js('The Gravedigger'), Js('The Gravekeeper'), Js('The Grim'), Js('The Grotesque'), Js('The Guardian'), Js('The Hallowed'), Js('The Hammer'), Js('The Hawk'), Js('The Heartless'), Js('The Hog'), Js('The Hollow'), Js('The Honorbound'), Js('The Honorless'), Js('The Horror'), Js('The Hound'), Js('The Hunger'), Js('The Hungry'), Js('The Hunter'), Js('The Hurricane'), Js('The Impure'), Js('The Insane'), Js('The Invincible'), Js('The Ironclad'), Js('The Ironfist'), Js('The Jackal'), Js('The Jester'), Js('The Legionnaire'), Js('The Limp'), Js('The Lion'), Js('The Lionheart'), Js('The Lone Wolf'), Js('The Magnificent'), Js('The Mammoth'), Js('The Maneater'), Js('The Maniac'), Js('The Marked'), Js('The Marked One'), Js('The Martyr'), Js('The Menace'), Js('The Merciful'), Js('The Miracle'), Js('The Monster'), Js('The Mountain'), Js('The Mutant'), Js('The Mute'), Js('The Nightmare'), Js('The Nightowl'), Js('The Noble'), Js('The Nobody'), Js('The Observer'), Js('The Owl'), Js('The Ox'), Js('The Paragon'), Js('The Patient'), Js('The Phantom'), Js('The Phoenix'), Js('The Prince'), Js('The Prodigy'), Js('The Pyro'), Js('The Rabid'), Js('The Raven'), Js('The Rebel'), Js('The Reckless'), Js('The Rider'), Js('The Rogue'), Js('The Sadist'), Js('The Sadistic'), Js('The Savage'), Js('The Scar'), Js('The Scarred One'), Js('The Sentinel'), Js('The Serpent'), Js('The Shadow'), Js('The Shady'), Js('The Shepherd'), Js('The Silencer'), Js('The Silent'), Js('The Silver One'), Js('The Skeleton'), Js('The Slayer'), Js('The Sleeper'), Js('The Spider'), Js('The Spook'), Js('The Storm'), Js('The Surgeon'), Js('The Swine'), Js('The Temper'), Js('The Tempest'), Js('The Terror'), Js('The Thief'), Js('The Thirst'), Js('The Thug'), Js('The Thunder'), Js('The Titan'), Js('The Tower'), Js('The Traitor'), Js('The Tyrant'), Js('The Undying'), Js('The Valiant'), Js('The Vengeful'), Js('The Venom Tongue'), Js('The Vermin'), Js('The Vicious'), Js('The Viper'), Js('The Vulture'), Js('The Whisper'), Js('The Widow Maker'), Js('The Wild'), Js('The Wildling'), Js('The Wolf'), Js('The Wonder'), Js('The Wyrm')]))
var.put('nm2', Js([Js('Battle'), Js('Bear'), Js('Blood'), Js('Bone'), Js('Boulder'), Js('Bright'), Js('Dark'), Js('Dead'), Js('Death'), Js('Demon'), Js('Doom'), Js('Dragon'), Js('Ember'), Js('Fire'), Js('Fist'), Js('Frost'), Js('Fuse'), Js('Giant'), Js('Gold'), Js('Gore'), Js('Grand'), Js('Great'), Js('Hell'), Js('Iron'), Js('Light'), Js('Mammoth'), Js('Molten'), Js('Night'), Js('Phoenix'), Js('Proud'), Js('Rage'), Js('Raven'), Js('Red'), Js('Rock'), Js('Rumble'), Js('Shadow'), Js('Sharp'), Js('Shield'), Js('Silent'), Js('Silver'), Js('Single'), Js('Skull'), Js('Spirit'), Js('Steel'), Js('Stone'), Js('Storm'), Js('Stout'), Js('Strong'), Js('Swift'), Js('Thunder'), Js('True'), Js('Void'), Js('War'), Js('Wild'), Js('Wolf')]))
var.put('nm3', Js([Js('bane'), Js('blade'), Js('blood'), Js('blow'), Js('bolt'), Js('bow'), Js('breaker'), Js('brow'), Js('chaser'), Js('claw'), Js('cleaver'), Js('crest'), Js('cut'), Js('eye'), Js('fang'), Js('fist'), Js('flayer'), Js('fury'), Js('gaze'), Js('grim'), Js('grimace'), Js('grip'), Js('hair'), Js('hallow'), Js('hammer'), Js('hand'), Js('head'), Js('heart'), Js('helm'), Js('hide'), Js('mane'), Js('mantle'), Js('might'), Js('pelt'), Js('rage'), Js('roar'), Js('scar'), Js('scream'), Js('shade'), Js('shadow'), Js('shield'), Js('shout'), Js('snarl'), Js('song'), Js('sorrow'), Js('stare'), Js('stride'), Js('strike'), Js('sword'), Js('sworn'), Js('talon'), Js('thorn'), Js('tongue'), Js('visage')]))
var.put('nm4', Js([Js(''), Js(''), Js(' ')]))
pass
pass


# Add lib to the module scope
warriorNicknames = var.to_python()